runtime/process.go

```diff
@@ -193,8 +194,56 @@ func (p *process) Resize(w, h int) error {
 	return err
 }
 
-func (p *process) ExitStatus() (int, error) {
+func (p *process) handleSigkilledShim(rst int, rerr error) (int, error) {
+	if rerr == nil || p.cmd == nil || p.cmd.Process == nil {
+		return rst, rerr
+	}
+
+	// Possible that the shim was SIGKILLED
+	e := unix.Kill(p.cmd.Process.Pid, 0)
+	if e != syscall.ESRCH {
+		return rst, rerr
+	}
+
+	// Ensure we got the shim ProcessState
+	<-p.cmdDoneCh
+
+	shimStatus := p.cmd.ProcessState.Sys().(syscall.WaitStatus)
+	if shimStatus.Signaled() && shimStatus.Signal() == syscall.SIGKILL {
+		logrus.Debugf("containerd: ExitStatus(container: %s, process: %s): shim was SIGKILL'ed reaping its child with pid %d", p.container.id, p.id, p.pid)
+
+		var status unix.WaitStatus
+		var rusage unix.Rusage
+		var wpid int
+
+		for wpid == 0 {
+			wpid, e = unix.Wait4(p.pid, &status, unix.WNOHANG, &rusage)
+			if e != nil {
+				logrus.Debugf("containerd: ExitStatus(container: %s, process: %s): Wait4(%d): %v", p.container.id, p.id, p.pid, rerr)
+				return rst, rerr
+			}
+		}
+
+		if wpid == p.pid {
+			rerr = nil
+			rst = 128 + int(shimStatus.Signal())
```

Comment:
I get `0x8D00` as the status. The lower 8 bits are 0 which normally means the process hasn't been signaled.